operator/e2e_testing/utils/shared_cluster.go

```diff
@@ -0,0 +1,479 @@
+// /*
+// Copyright 2025 The Grove Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+package utils
```

Comment:
This is not really a utility but it is part of the setup for all e2e tests and should be moved to tests/setup package

Reply:
Hmm... that makes sense. This logic applies to everything else in this package though. I'll just rename it.